generators/dart/text.js

```diff
@@ -295,3 +295,50 @@ Blockly.Dart['text_prompt_ext'] = function(block) {
 };
 
 Blockly.Dart['text_prompt'] = Blockly.Dart['text_prompt_ext'];
+
+Blockly.Dart['text_count'] = function(block) {
+  var text = Blockly.Dart.valueToCode(block, 'TEXT',
+      Blockly.Dart.ORDER_UNARY_POSTFIX) || '\'\'';
+  var sub = Blockly.Dart.valueToCode(block, 'SUB',
+      Blockly.Dart.ORDER_NONE) || '\'\'';
+  // Substring count is not a native Dart function.  Define one.
+  var functionName = Blockly.Dart.provideFunction_(
+      'text_count',
+      ['int ' + Blockly.Dart.FUNCTION_NAME_PLACEHOLDER_ +
+        '(String string, String search) {',
+        '  int index = 0;',
+        '  int count = 0;',
+        '  while (index != -1) {',
+        '    index = string.indexOf(search, index);',
+        '    if (index != -1) {',
+        '      count++;',
+        '     index += search.length;',
+        '    }',
+        '  }',
+        '  return count;',
+        '}']);
+  var code = functionName + '(' + text + ', ' + sub + ')';
+  return [code, Blockly.Dart.ORDER_UNARY_POSTFIX];
+};
+
+Blockly.Dart['text_replace'] = function(block) {
+  var text = Blockly.Dart.valueToCode(block, 'TEXT',
+      Blockly.Dart.ORDER_UNARY_POSTFIX) || '\'\'';
+  var from = Blockly.Dart.valueToCode(block, 'FROM',
+      Blockly.Dart.ORDER_NONE) || '\'\'';
+  var to = Blockly.Dart.valueToCode(block, 'TO',
+      Blockly.Dart.ORDER_NONE) || '\'\'';
+  var code = text + '.replaceAll(' + from + ', ' + to + ')';
+  return [code, Blockly.Dart.ORDER_UNARY_POSTFIX];
+};
+
+Blockly.Dart['text_reverse'] = function(block) {
+  // There isn't a sensible way to do this in Dart. See:
+  // http://stackoverflow.com/a/21613700/3529104
+  // Implementing something is possibly better than not implementing anything?
+  var text = Blockly.Dart.valueToCode(block, 'TEXT',
+      Blockly.Dart.ORDER_UNARY_POSTFIX) || '\'\'';
+  var code = 'new String.fromCharCodes(' + text + '.runes.toList().reversed)';
+  // XXX What should the operator precedence be for a `new`?
```

Comment:
I don't know Dart at all, and the language reference didn't yield any obvious answer to this question. `ORDER_UNARY_POSTFIX` looks like it won't break anything, even though it might result in some unnecessary parens.